src/clip.js

```diff
@@ -41,13 +41,17 @@ function clip(features, scale, k1, k2, axis, intersect, minAll, maxAll) {
         if (slices.length) {
             // if a feature got clipped, it will likely get clipped on the next zoom level as well,
             // so there's no need to recalculate bboxes
-            clipped.push({
+            var clippedFeature = {
                 geometry: slices,
                 type: type,
                 tags: features[i].tags || null,
                 min: feature.min,
                 max: feature.max
-            });
+            };
+            if (typeof feature.id !== 'undefined') {
+                clippedFeature.id = feature.id;
+            }
```

Comment:
In the context of https://github.com/mapbox/mapbox-gl-js/issues/2874 what if a feature has no id provided, or any properties for that matter, it would be good to still be able to do feature highlights on hover in this case. Would this require geojson-vt to generate ids where the don't exist so that GL JS can use them in a hover filter?